systemdev/testing-locally.rst

```diff
@@ -3,18 +3,203 @@ Making changes and testing locally
+
+We'll examine the use of crossbuilder and builds on the device using `address-book-app <https://github.com/ubports/address-book-app>`__ (the Contacts application) as an example.
 
 We only recommend developing packages using a device with Ubuntu Touch installed from the devel channel. This ensures that you are testing your changes against the most current state of the Ubuntu Touch code.
 
-.. warning::
+Preparing a package for a build
+-------------------------------
 
-    Installing packages has a risk of damaging the software on your device, rendering it unusable. If this happens, you can :doc:`reinstall Ubuntu Touch </userguide/install>`.
+``sbuild`` or ``crossbuilder`` need to be invoked from a debianized package source tree (i.e. the package sources with a ``debian`` subdirectory), UBports packages mostly consists of git repositories containing native or non-native packages.  Native packages can be built directly and the following script can be used in order to prepare a non-native package for a build::
+
+    #!/bin/sh
+
+    PATH=/bin:/usr/bin
+
+    die() {
+        if [ $# -gt 0 ]; then
+            printf "%s\n" "$1" >&2
+        fi
+        exit 1
+    }
+
+    if [ ! -d "./debian" ]; then
+        die "not in a debianized package directory"
+    fi
+
+    if [ -f "./debian/ubports.source_location" ]; then
+
+        {
+            read -r src_url && \
+            read -r src_filename
+        } < "./debian/ubports.source_location" || \
+            die "failed to parse ubports.source_location"
+        case ${src_url} in
+        http://*|https://*|ftp://*)
+            ;;
+        *)
+            die "invalid url: \"${src_url}\""
+            ;;
+        esac
+        src_filename="$(basename "${src_filename}")"
+
+        wget -O "../${src_filename}" "${src_url}" || \
+            die "failed to download source archive"
+    fi
+
+
+Building packages in a chroot using sbuild
+------------------------------------------
+
+`sbuild <https://tracker.debian.org/pkg/sbuild>`__ is a tool for building Debian packages from source in an isolated environment using a chroot created by `schroot <https://wiki.debian.org/Schroot>`__. It closely resembles the package build process on the UBports CI system by using an isolated build environment with a minimal set of pre-installed packages. This detects any missing build dependencies and sbuild will also detect problems with the resulting packages by running `lintian <https://lintian.debian.org/manual/lintian.html>`__
+
+Prerequisites
+^^^^^^^^^^^^^
+
+A host system running either Debian 11 (Bullseye) or later or Ubuntu 20.04 (Focal Fossa) or later is required.
+An easy and performant way to set up Debian or Ubuntu on other distributions is to run it in a container using `LXD <https://linuxcontainers.org/lxd/getting-started-cli/#installing-a-package>`.
```

Comment:
link doesn't render

Reply:
Works fine for me as well, maybe you're affected by the Firefox issue?

Reply:
Lol. I was most definitely affected by the Firefoxalypse :) But, I'm also quite sure that these two links don't render correctly. Have you checked the CI build I linked?

![image](https://user-images.githubusercontent.com/26801304/149621335-8c2201be-4733-4e52-82d8-1bb04e217010.png)

I think with the first link it's the underscores that are *inside* the backticks, with the second link the underscores are missing.

Reply:
Thanks for clarifying, should be fixed now.